sys/posix/sockets/posix_sockets.c

```diff
@@ -857,12 +858,23 @@ ssize_t sendto(int socket, const void *buffer, size_t length, int flags,
                 /* cppcheck bug? res is read below in l824 */
                 /* cppcheck-suppress unreadVariable */
                 res = conn_udp_sendto(buffer, length, src_addr, src_len, addr, addr_len, s->domain,
-                                      sport, port);
+                                      sport, byteorder_ntohs(port));
             }
             else if (address != NULL) {
-                uint16_t sport = (uint16_t)genrand_uint32_range(1LU << 10U, 1LU << 16U);
+                ipv6_addr_t local;
+                if (s->src_port == 0) {
```

Comment:
is this check really necessary? `s->bound` evaluates to `false` when we hit this line